colcon_core/environment/__init__.py

```diff
@@ -126,8 +126,12 @@ def create_environment_scripts(
     # create file containing the runtime dependencies
     path = prefix_path / get_relative_package_index_path() / pkg.name
     path.parent.mkdir(parents=True, exist_ok=True)
+    dependencies = [dependency.name
+                    for dependency in pkg.dependencies.get('run', set())]
+    # TODO: Is this supposed to be in topological order? I don't think the
+    # set is ordered?
     path.write_text(
-        os.pathsep.join(sorted(pkg.dependencies.get('run', set()))))
+        os.pathsep.join(sorted(dependencies)))
```

Comment:
This functions docstring states:     

:returns: iterable of the generated hook paths
:rtype: Iterable

But there is no return statement, is that an issue?

Reply:
I think the docstring is just wrong. Please address this in a separate PR (otherwise I can do that too if you prefer).